yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -751,6 +785,43 @@ export class LibP2PService<T extends P2PClientType = P2PClientType.Full> extends
     );
   }
 
+  private async extractTxsFromBlockProposal(block: BlockProposal, peerId: PeerId): Promise<void> {
+    const receivedTxs = block.txs ?? [];
+    const receivedTxHashes = await Promise.all(receivedTxs.map(tx => tx.getTxHash()));
+    if (receivedTxHashes.length === 0) {
+      return;
+    }
+    // Check if we already have the transactions in our pool
+    const haveAlready = await this.mempools.txPool.hasTxs(receivedTxHashes);
+    const newTxs = receivedTxs.filter((_, i) => !haveAlready[i]);
+    if (newTxs.length === 0) {
+      return;
+    }
+    // Validate the transactions
+    const blockNumber = (await this.archiver.getBlockNumber()) + 1;
```

Comment:
```suggestion
    const blockNumber = block.blockNumber;
```
It should be the same if the archiver is synced, but just in case.